xarray/core/_typed_ops.py

```diff
@@ -632,6 +632,8 @@ def __eq__(self, other: VarCompatible) -> Self:
     def __eq__(self, other: VarCompatible) -> Self | T_DataArray:
         return self._binary_op(other, nputils.array_eq)
 
+    __hash__: None  # type:ignore[assignment]
```

Comment:
Good for testing but FYI: https://github.com/pydata/xarray/blob/ec1b67e8efd1f53500ec351a11ff216c4e332704/xarray/core/_typed_ops.py#L2